.githooks/pre-push

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Wire
+# Copyright (C) 2023 Wire Swiss GmbH
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program. If not, see http://www.gnu.org/licenses/.
+#
+
+set -e
+
+echo "-----------------------------------------------------------"
+echo "> Running detekt checks locally before push..."
+echo "-----------------------------------------------------------"
+
+./gradlew clean detekt
```

Comment:
Really needed?

Annoying when working offline, and Detekt is already running before tests anyway, so CI time is safe.

We are grown up and we should be responsible for our workflow.
If people aren't proactive enough to run Detekt locally before pushing, and prefer going back and forward seeing CI output instead of checking first.
I don't think we should babysit.

Sometimes I do forget about it, and I pay the price.
To me, having to run `detekt` on commits where I just changed some SQL or some other stuff that won't be detected would be more detrimental.

Personally, I'll disable the hook on my IDE.

Reply:
Yep yep! Totally agree, we are grown-ups to take care of this, as a hook running locally of course, we have total control to disable it or just ignore it with `--no-verify` that's the beauty of git, everything is possible 🧽 🙌 🌈 

I think is more of a misunderstanding on this particular hook itself, this will enforce no one to use it, it's more of an _opt-in_ feature if we want to look at it, being, if no one wants to install or ever run the `make setup/pre-push-hook` the hook will be living there for free on the repo, with 0 usages, so nothing to disable =)